frontend/src/pages/spotnet/dashboard/Dashboard.jsx

```diff
@@ -1,124 +1,20 @@
-    if (!walletId) {
-      console.error('closePositionEvent: walletId is undefined');
-      return;
-    }
-    try {
-      const response = await axiosInstance.get(`/api/get-repay-data?supply_token=ETH&wallet_id=${walletId}`);
-      await closePosition(response.data);
-
-      await axiosInstance.get(`/api/close-position?position_id=${response.data.position_id}`);
-    } catch (e) {
-      console.error('Error during closePositionEvent', e);
-    }
-  };
-
-  const [cardData, setCardData] = useState([
-    {
-      title: 'Collateral & Earnings',
-      icon: CollateralIcon,
-      balance: '0.00',
-      currencyName: 'Ethereum',
-      currencyIcon: EthIcon,
-    },
-    {
-      title: 'Borrow',
-      icon: BorrowIcon,
-      balance: '0.00',
-      currencyName: 'USD Coin',
-      currencyIcon: UsdIcon,
-    },
-  ]);
-
-  const [startSum] = useState(0);
-  const [currentSum, setCurrentSum] = useState(0); 
-  const [healthFactor, setHealthFactor] = useState('0.00');
-  const [loading, setLoading] = useState(true);
-
   const starData = [
     { top: 1, left: 0, size: 1.5 },
     { top: 75, left: 35, size: 2.5 },
     { top: -2, left: 94, size: 5.5 },
   ];
 
-  useEffect(() => {
-    const getData = async () => {
-      if (!walletId) {
-        console.error('getData: walletId is undefined');
-        setLoading(false);
-        return;
-      }
-
-      const data = await fetchCardData({ walletId });
-      if (data && data.zklend_position && data.zklend_position.products) {
-        const positions = data.zklend_position.products[0].positions || [];
-        const healthRatio = data.zklend_position.products[0].health_ratio;
-
-        const cardData = positions.map((position, index) => {
-          const isFirstCard = index === 0;
-          const tokenAddress = position.tokenAddress;
-
-          if (isFirstCard) {
-            const isEthereum = tokenAddress === ZETH_ADDRESS;
-            const balance = parseFloat(position.totalBalances[Object.keys(position.totalBalances)[0]]);
-            setCurrentSum(balance);
-
-            return {
-              title: 'Collateral & Earnings',
-              icon: CollateralIcon,
-              balance: balance,
-              currencyName: isEthereum ? 'Ethereum' : 'STRK',
-              currencyIcon: isEthereum ? EthIcon : StrkIcon,
-            };
-          }
-
-          return {
-            title: 'Borrow',
-            icon: BorrowIcon,
-            balance: position.totalBalances[Object.keys(position.totalBalances)[0]],
-            currencyName: 'USD Coin',
-            currencyIcon: UsdIcon,
-          };
-        });
-
-        setCardData(cardData);
-        setHealthFactor(healthRatio);
-      } else {
-        console.error('Data is missing or incorrectly formatted');
-      }
-      setLoading(false);
-    };
-
-    getData();
-  }, [walletId]);
+  const [startSum] = useState(0);
+  const { cardData, healthFactor, currentSum, isLoading } = useDashboardData(walletId);
```

Comment:
I don't think that `isLoading` is returned from `useDashboardData`. My assumption it is undefined, if you want loading state from useQuery, `isPending` is valid.